native/pca_bench.cpp

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (C) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <vector>
+#include <utility>
+#include <algorithm>
+#include <iostream>
+#include <chrono>  
+
+#include "common.hpp"
+#include "daal.h"
+
+
+namespace dn = daal::algorithms::normalization;
+
+
+std::pair<da::pca::ResultPtr, dm::NumericTablePtr>
+pca_fit_daal(double *X, size_t rows, size_t cols, size_t n_components) {
+
+    // Find number of components to use in DAAL
+    if (n_components < 1) {
+        n_components = std::min(n_components, rows);
```

Comment:
Did you mean to write `n_components = std::min(cols, rows)`  instead?

Reply:
Done!